lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -0,0 +1,214 @@
+import pytest
+import requests
+from http import HTTPStatus
+
+
+class Commons:
+    """
+    Unit testing for all the elasticsearch resources class.
+
+    In a web service context, we access class functions mostly via the
+    Flask test client rather than trying to directly invoke the class
+    constructor and `post` service.
+    """
+
+    def build_index(self, server_config, dates):
+        """
+        Build the index list for query
+
+        Args:
+            dates (iterable): list of date strings
+        """
+        idx = server_config.get("Indexing", "index_prefix") + ".v6.run-data."
+        index = "/"
+        for d in dates:
+            index += f"{idx}{d},"
+        return index
+
+    def test_non_accessible_user_data(self, client, server_config, pbench_token):
+        """
+        Test behavior when Authorization header does not have access to other user's data
+        """
+        # The pbench_token fixture logs in as user "drb"
+        # Trying to access the data belong to the user "pp"
+        self.json["user"] = "pp"
+        response = client.post(
+            f"{server_config.rest_uri}{self.pbench_endpoint}",
```

Comment:
That doesn't quite make sense:  these methods are _instance_ methods, so there _is_ an object instantiation happening, perhaps in the background, and so there is definitely an opportunity for a constructor to be _called_...now, it might be the case that PyTest is itself generating and/or taking over that function, which is entirely possible, but it would be a violation of Python Language semantics to imply that it's not there!

In any case, if you don't want to do it via a c'tor, that's fine -- the base class can define a `_setup()` function (if that name is not also reserved to PyTest), which the subclasses can call -- the point is, the superclass really should not be depending on attributes defined in the subclasses, and subclasses really should depend on the superclass to initialize the attributes which it defines and depends on.

Reply:
Yes I didnt mean that the object instantiation isn't happening its just that the pytest ignores any classes defined with `__init__` https://doc.pytest.org/en/latest/explanation/goodpractices.html#conventions-for-python-test-discovery

pytest does instantiate the class in order to execute the tests but if the constructor takes any arguments it can't instantiate it. However, yes I will experiment with `_setup` function and see if that behaves like init for us

Reply:
Thanks for the documentation link (I should have looked that up myself, yesterday, but I was short on time).

PyTest needs to instantiate the class in order to execute the tests, but, if the c'tor were to require arguments, then PyTest wouldn't have any values to provide, so they address the problem by simply skipping any class which provides _any_ c'tor.  That seems a little heavy-handed to me, but, OK...our initialization function doesn't _have_ to be called `__init__()` and it doesn't _have_ to be called implicitly, so we should be good.

Reply:
You've addressed this in #2413.